src/CommonITILValidation.php

```diff
@@ -1993,4 +1947,152 @@ public static function getAllValidationStatusArray()
+            $vs = new ValidationStep();
+            if (!$vs->getFromDB($validationsteps_id)) {
+                Session::addMessageAfterRedirect('Failed to load approval step while adding approval step.');
+            };
+
+            // create ITIL_ValidationStep
+            $itil_class = static::getItilObjectItemType();
+            $itil_validationstep = $itil_class::getValidationStepInstance();
+            $itil_validationstep->add([
+                'validationsteps_id' => $validationsteps_id,
+                'minimal_required_validation_percent' => $vs->fields['minimal_required_validation_percent'],
+            ]);
+
+            $this->input['itils_validationsteps_id'] = $itil_validationstep->getID();
+        }
+
+        unset($this->input['_validationsteps_id']);
+    }
+
+    /**
+     * Delete, only if the itils_validationstep is not used anymore
+     *
+     * @param int $itils_validationsteps_id
+     * @return void
+     */
+    private function removeUnsedITILValidationStep(int $itils_validationsteps_id): void
+    {
+        /** @var \CommonITILObject $itil */
+        $itil = $this->getItem();
+        $validation = $itil::getValidationClassInstance();
+        $validations = $validation->find(['itils_validationsteps_id' => $itils_validationsteps_id]);
+        if (!empty($validations)) {
+            // itils_validation is still used, do not delete
+            return;
+        }
+
+        if (!$itil::getValidationStepInstance()->delete(['id' => $itils_validationsteps_id])) {
+            Session::addMessageAfterRedirect('Failed to delete unused approval step.');
+        };
+    }
+
+    public function recomputeItilStatus(): void
+    {
+        /** @var CommonITILObject $itil */
+        $itil = $this->getItem();
+        if (
+            !$itil->update([
+                'id' => $itil->getID(),
+                'global_validation' => self::computeValidationStatus($itil),
+                '_from_itilvalidation' => true,
+            ])
+        ) {
+            Session::addMessageAfterRedirect('Failed to update Itil global approval status.');
+        }
+    }
+
+    /**
+     * Update minimal required validation percent of ITILValidationStep
+     */
+    private function updateITILValidationStepThreshold(int $threshold): void
+    {
+        $itil_validationstep = new (static::$itemtype::getValidationStepClassName());
+        if (!$itil_validationstep->getFromDB($this->fields['itils_validationsteps_id'])) {
+            throw new \RuntimeException('Invalid ITIL validation step. ' . $this->fields['itils_validationsteps_id']);
+        };
+        if ((int) $itil_validationstep->fields['minimal_required_validation_percent'] != $threshold) {
+            if (!$itil_validationstep->update(['id' => $this->fields['itils_validationsteps_id'], 'minimal_required_validation_percent' => $threshold])) {
+                Session::addMessageAfterRedirect('Failed to update approval step threshold.');
```

Comment:
`addMessageAfterRedirect` defaults to an INFO type message. Failures should be at least WARNING level.